### PR TITLE
drivers: Add compile configurations groups and add parameters

### PR DESCRIPTION
**Pull request by akshaim**
### Contribution description

1. Create Subgroups in drivers/doc.txt

Identify Compile Time Parameters in  the following drivers 

2. drivers/include/hdc1000.h
3. drivers/include/ili9341.h
4. drivers/include/kw2xrf.h
5. drivers/include/kw41zrf.h
6. drivers/include/mag3110.h
7. drivers/include/motor_driver.h
8. drivers/include/mpl3115a2.h
9. drivers/include/mtd_sdcard.h
10. drivers/include/opt3001.h
11. drivers/include/pir.h
12. drivers/include/pn532.h
13. drivers/include/rn2xx3.h
14. drivers/include/slipdev.h
15. drivers/include/soft_spi.h
16. drivers/include/sps30.h
17. drivers/include/tcs37727.h
18. drivers/include/tmp00x.h

29.drivers/include/mma8x5x.h

Add to subgroups :

19. drivers/include/mrf24j40.h : Add to config_drivers_netdev group
20. drivers/include/sx127x.h : Add to config_drivers_netdev group 
21. drivers/adcxx1c/include/adcxx1c_params.h : Add to config_drivers_sensors group
22. drivers/include/at.h : Add to config_drivers_misc group
23. drivers/include/at86rf215.h : Add to config_drivers_netdev group
24. cpu/nrf52/include/nrf802154.h : Add to config_drivers_netdev group
25. drivers/include/dose.h : Add to config_drivers_netdev group 
26. drivers/include/ds1307.h : Add to config_drivers_sensors group
27. drivers/include/fxos8700.h : Add to config_drivers_sensors group
28. drivers/include/ads101x.h : Add to config_drivers_sensors group

Skipped the following Parameters 

1. drivers/include/ds1307.h-> I2C Address remains constant.
2. drivers/include/hts221.h-> I2C Address remains constant.
3. ~~drivers/include/mma8x5x.h -> I2C Address remains constant.~~
4. drivers/include/mpl3115a2.h  -> I2C Address remains constant.
5. drivers/include/tcs37727.h -> I2C Address remains constant. 
6. drivers/include/tmp00x.h - > Skipped calibration constant

### Testing procedure

Doxygen build works fine.

### Issues/PRs references

#10566

### CODE REVIEW

**Comment by akshaim** [comment]
@leandrolanzieri  Please review

**Comment by akshaim** [comment]
Thank you for the comments @leandrolanzieri  I shall make the changes now.

File: drivers/include/sps30.h [hunk 1]
```diff
@@ -128,7 +128,13 @@ typedef enum {
...
 
 /**
- * @brief     Maximum number of automatic retries on communication errors
+ * @defgroup drivers_SPS30_config     SPS30 PM Sensor driver compile configuration
+ * @ingroup config_sensors
+ * @{
+ */
+
+/**
+ * @brief     Set maximum number of automatic retries on communication errors
```
Comment (leandrolanzieri) [inline drivers/include/sps30.h:137]:
```suggestion
 * @brief     Maximum number of automatic retries on communication errors
```

File: drivers/include/tcs37727.h [hunk 2]
```diff
@@ -40,9 +40,19 @@ extern "C"
...
 #endif
 
+/**
+ * @defgroup drivers_tcs37727_config     TCS37727 RGB Light Sensor driver compile configuration
+ * @ingroup config_sensors
+ * @{
+ */
+
+/**
+ * @brief Set the default RGBC integration time.
```
Comment (leandrolanzieri) [inline drivers/include/tcs37727.h:50]:
```suggestion
 * @brief Default RGBC integration time.
```
Also, now that we are changing this, can we add the units to the doc?

File: drivers/include/tmp00x.h [hunk 3]
```diff
@@ -93,21 +93,30 @@ extern "C"
...
- * @brief   TMP00X Default Address
+ * @defgroup drivers_tmp00x_config     TMP006/TMP007 Infrared Thermopile Sensor driver compile configuration
+ * @ingroup config_sensors
+ * @{
+ */
+
+/**
+ * @brief   Set the Default Address
+ * 
+  * I2C Address depends on the state of ADR0 and ADR1 Pins
```
Comment (leandrolanzieri) [inline drivers/include/tmp00x.h:104]:
```suggestion
 * I2C Address depends on the state of ADR0 and ADR1 Pins
```

File: drivers/include/tmp00x.h [hunk 4]
```diff
@@ -93,21 +93,30 @@ extern "C"
...
- * @brief   Default Conversion Time in us
+ * @brief   Set the default Conversion Time in us
  */
 #ifndef TMP00X_CONVERSION_TIME
 #define TMP00X_CONVERSION_TIME     (1E6)
 #endif
 
 /**
- * @brief   Default low power mode
+ * @brief   Set the default low power mode mode
```
Comment (leandrolanzieri) [inline drivers/include/tmp00x.h:119]: Repeated `mode`

File: drivers/include/tmp00x.h [hunk 5]
```diff
@@ -93,21 +93,30 @@ extern "C"
...
 
 /**
- * @brief   TMP00X Default Address
+ * @defgroup drivers_tmp00x_config     TMP006/TMP007 Infrared Thermopile Sensor driver compile configuration
+ * @ingroup config_sensors
+ * @{
+ */
+
+/**
+ * @brief   Set the Default Address
```
Comment (leandrolanzieri) [inline drivers/include/tmp00x.h:102]: Here and bellow, please remove `Set the`.

File: drivers/include/hdc1000.h [hunk 6]
```diff
@@ -51,8 +51,17 @@ extern "C"
 {
 #endif
 
+/**
```
[the remark this answers is not in the document]
Comment (akshaim) [reply drivers/include/hdc1000.h:54]: Yes. Should I remove the other PR ?

[the remark this answers is not in the document]
Comment (leandrolanzieri) [reply drivers/include/hdc1000.h:54]: Yes, you can close it

**Comment by akshaim** [comment]
Done. Please review @leandrolanzieri

**Comment by akshaim** [comment]
Thank you @leandrolanzieri  for th comments. Please see if this is okay.

**Comment by leandrolanzieri** [comment]
@akshaim this looks good already. Just some details:
- The configuration group for the ADS101 still shows up on `config`, it should be included in the sensors subgroup.
- Please remove `/../..` from the commit message of c579f978db69c86f0c0f703bb368ed080a8fc731
In general having `drivers/<driver>` should be enough to understand the scope of the change.

**Comment by akshaim** [comment]
@leandrolanzieri  Please see if this is okay. If fine I shall do a rebase.

File: drivers/include/ethos.h [hunk 7]
```diff
@@ -34,14 +34,27 @@ extern "C" {
 #endif
 
 /* if using ethos + stdio, use STDIO_UART values unless overridden */
-#ifdef MODULE_STDIO_ETHOS
+#ifdef MODULE_STDIO_ETHOS || DOXYGEN
```
Comment (leandrolanzieri) [inline drivers/include/ethos.h:37]: The CI pointed out that this should be:
```suggestion
#if IS_USED(MODULE_STDIO_ETHOS) || defined(DOXYGEN)
```
You will need to `#include "kernel_defines.h"` to use `IS_USED` macro

Comment (akshaim) [reply drivers/include/ethos.h]: Updates. Please check

**Comment by leandrolanzieri** [comment]
It works. @akshaim please squash!

**Comment by akshaim** [comment]
Done.  @leandrolanzieri

**Comment by akshaim** [comment]
@leandrolanzieri  Added drivers/mma8x5x.h as per your suggestion from #13831

**Comment by leandrolanzieri** [comment]
Looks good. Please squash @akshaim

**Comment by akshaim** [comment]
D

> Looks good. Please squash @akshaim

Done

**Comment by akshaim** [comment]
Thanks @leandrolanzieri :)

**Pull request merged**